lib/analyzer.h

```diff
@@ -129,6 +129,8 @@ struct Analyzer {
     virtual void forkScope(const Token* /*endBlock*/) {}
     /// If the value is conditional
     virtual bool isConditional() const = 0;
+    /// If the value is known
+    virtual bool isKnown() const = 0;
```

Comment:
If `isConditional` is not sufficient(thats what we use for if/else branches), then I think this should be named by where this function is used(like `isSwitchable` for lack of a better name). This will let use extend the logic beyond `isKnown`. Perhaps we want to allow traversing switches the value isKnown and isImpossible. Lifetime values also may want to traverse switched when the value isPossible(as it doesn't make any lifetimes known).

Although, I think it would be better to use `isConditional` for this.

Reply:
We could also use `updateScope` method instead of `isConditional`. This might be a better option.